apis/batch/v1alpha1/job_types.go

```diff
@@ -0,0 +1,304 @@
+	//  [LogEntry](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry)
+	//  that it generates.
+	//
+	//  Use labels to group and describe the resources they are applied to. Batch
+	//  automatically applies predefined labels and supports multiple `labels`
+	//  fields for each job, which each let you apply custom labels to various
+	//  resources. Label names that start with "goog-" or "google-" are
+	//  reserved for predefined labels. For more information about labels with
+	//  Batch, see
+	//  [Organize resources using
+	//  labels](https://cloud.google.com/batch/docs/organize-resources-using-labels).
+	// +kcc:proto:field=google.cloud.batch.v1.Job.labels
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Log preservation policy for the Job.
+	// +kcc:proto:field=google.cloud.batch.v1.Job.logs_policy
+	LogsPolicy *LogsPolicy `json:"logsPolicy,omitempty"`
+
+	// Notification configurations.
+	// +kcc:proto:field=google.cloud.batch.v1.Job.notifications
+	Notifications []JobNotification `json:"notifications,omitempty"`
+
+	// Required. The parent resource name where the Job will be created. Pattern: "projects/{project}/locations/{location}"
+	*Parent `json:",inline"`
+
+	// The BatchJob name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
+}
+
+type Parent struct {
+	// Immutable. The location where the alloydb cluster should reside.
+	// +required
+	Location *string `json:"location,omitempty"`
+
+	// The project that this resource belongs to.
+	// +required
+	ProjectRef *v1beta1.ProjectRef `json:"projectRef,omitempty"`
+}
+
+// +kcc:proto=google.cloud.batch.v1.AllocationPolicy
+type AllocationPolicy struct {
+	// Location where compute resources should be allocated for the Job.
+	// +kcc:proto:field=google.cloud.batch.v1.AllocationPolicy.location
+	Location *AllocationPolicy_LocationPolicy `json:"location,omitempty"`
+
+	// Describe instances that can be created by this AllocationPolicy.
+	//  Only instances[0] is supported now.
+	// +kcc:proto:field=google.cloud.batch.v1.AllocationPolicy.instances
+	Instances []AllocationPolicy_InstancePolicyOrTemplate `json:"instances,omitempty"`
+
+	// Defines the service account for Batch-created VMs. If omitted, the [default
+	//  Compute Engine service
+	//  account](https://cloud.google.com/compute/docs/access/service-accounts#default_service_account)
+	//  is used. Must match the service account specified in any used instance
+	//  template configured in the Batch job.
+	//
+	//  Includes the following fields:
+	//   * email: The service account's email address. If not set, the default
+	//   Compute Engine service account is used.
+	//   * scopes: Additional OAuth scopes to grant the service account, beyond the
+	//   default cloud-platform scope. (list of strings)
+	// +kcc:proto:field=google.cloud.batch.v1.AllocationPolicy.service_account
+	ServiceAccount *v1beta1.IAMServiceAccountRef `json:"serviceAccount,omitempty"`
```

Comment:
When we change a field to a reference, the json and go field names are normally changed to <field>Ref. So this would be ServiceAccountRef and `serviceAccountRef`